services/payments/controllers.go

```diff
@@ -84,7 +107,7 @@ func PrepareHandler(service *Service) handlers.AppHandler {
 		}
 
 		// returns an enriched list of transactions, which includes the document metadata
-		resp, err := service.PrepareTransaction(ctx, req)
+		resp, err := service.PrepareTransaction(ctx, namespaceUUID, req)
```

Comment:
it seems weird to pass this as an argument. would it make more sense to be a property of the service?